File: week_01_arrays/1.1.py
'''
Push Front
Given array and an additional value, insert this value at the beginning of the array.
'''
def push_front(arr, val):
    # Easiest:
    # arr.insert(0,val)
    # return arr

    # Duplicate final value of array
    arr.append(arr[-1])
    for idx in reversed(range(0,len(arr))):
        arr[idx] = arr[idx-1]
    arr[0] = val
    return arr

print push_front([7,5,3],3)

File: week_01_arrays/1.1.js
/*
Push Front
Given array and an additional value, insert this value at the beginning of the array. Do this without using any built-in array methods.
*/

function PushFront(arr, val){
  // Easiest:
  // arr.unshift(val)
  // return arr

  for (var i = arr.length; i > 0; i-=1) {
    arr[i] = arr[i-1];
  }
  arr[0] = val;
  return arr;
}

// console.log(PushFront([3,4,5],6))
